ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -18657,20 +18657,52 @@ void HqlCppTranslator::doBuildNewRegexFindReplace(BuildCtx & ctx, const CHqlBoun
     // as long as the find instance.  Only exception could be if call created a temporary class instance.
     if (expr->getOperator() == no_regex_replace)
     {
-        HqlExprArray args;
-        args.append(*LINK(compiled));
-        args.append(*LINK(search));
-        args.append(*LINK(expr->queryChild(2)));
-        IIdAtom * func = nullptr;
-        if (isUTF8Type(searchStringType))
-            func = regexNewU8StrReplaceXId;
-        else if (isUnicodeType(searchStringType))
-            func = regexNewUStrReplaceXId;
+        if (target && target->isFixedSize())
```

Comment:
This test needs to be more strict.  It should check 

```
&& target->queryType()->getTypeCode() == expr->queryType()->getTypeCode()
```

Otherwise a regexfind that is cast to an integer will match the condition.  One example in fromxml6.eclxml(~6854)

I hit the  throwUnexpectedType() in getElementPointer.

Reply:
Excellent catch; fixed, with comment for posterity.